providers/unleash/src/main/java/dev/openfeature/contrib/providers/unleash/ContextTransformer.java

```diff
@@ -0,0 +1,75 @@
+package dev.openfeature.contrib.providers.unleash;
+
+import dev.openfeature.sdk.EvaluationContext;
+import dev.openfeature.sdk.ImmutableContext;
+import dev.openfeature.sdk.Value;
+import io.getunleash.UnleashContext;
+
+import java.time.ZonedDateTime;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Transformer from Unleash context to OpenFeature context and vice versa.
+ */
+public class ContextTransformer {
+
+    public static final String CONTEXT_APP_NAME = "appName";
+    public static final String CONTEXT_USER_ID = "userId";
+    public static final String CONTEXT_ENVIRONMENT = "environment";
+    public static final String CONTEXT_REMOTE_ADDRESS = "remoteAddress";
+    public static final String CONTEXT_SESSION_ID = "sessionId";
+    public static final String CONTEXT_CURRENT_TIME = "currentTime";
+
+    protected static UnleashContext transform(EvaluationContext ctx) {
+        UnleashContext.Builder unleashContextBuilder = new UnleashContext.Builder();
+        ctx.asObjectMap().forEach((k, v) -> {
+            switch (k) {
+                case CONTEXT_APP_NAME:
+                    unleashContextBuilder.appName(String.valueOf(v));
+                    break;
+                case CONTEXT_USER_ID:
+                    unleashContextBuilder.userId(String.valueOf(v));
+                    break;
+                case CONTEXT_ENVIRONMENT:
+                    unleashContextBuilder.environment(String.valueOf(v));
+                    break;
+                case CONTEXT_REMOTE_ADDRESS:
+                    unleashContextBuilder.remoteAddress(String.valueOf(v));
+                    break;
+                case CONTEXT_SESSION_ID:
+                    unleashContextBuilder.sessionId(String.valueOf(v));
+                    break;
+                case CONTEXT_CURRENT_TIME:
+                    unleashContextBuilder.currentTime(ZonedDateTime.parse(String.valueOf(v)));
```

Comment:
I would like to see some more tests around this one in particular. How are the user supposed to set the field?

Reply:
It is included and verified in [test](https://github.com/open-feature/java-sdk-contrib/blob/8a6316705d8ac7779ad2f00db4b18710120de7d0/providers/unleash/src/test/java/dev/openfeature/contrib/providers/unleash/UnleashProviderTest.java#L216).